client.go

```diff
@@ -379,6 +383,19 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return offset, err
 }
 
+func (client *client) Controller() (*Broker, error) {
+	if client.Closed() {
+		return nil, ErrClosedClient
+	}
+
+	controller := client.cachedController()
+	if controller == nil {
+		return nil, ErrControllerNotAvailable
```

Comment:
Rather than returning an error immediately we should try refreshing the metadata once to see if we can fetch the controller - being nil may just mean `RefreshMetadata` has never been run.

Reply:
Ok, I will work on it. But it's a little complicated，I need separating a new method `refreshMetadata` from `backgroundMetadataUpdater` to refresh the meta data because of the configuration `Metadata.Full`.